### ISSUE TITLE
Double mediation binding

### ISSUE THREAD

**Issue by mohaider** [strangeioc/strangeioc#178]
Hello gentlemen. 
I have been tinkering around StrangeIoC for a month and a half now, it's been such a pleasurable experience so far. Sorry if this isn't the correct place to bring an behavior I've been experiencing with strange, I saw your google code page and didn't want to post there as google code is dying. 
I am currently experiencing a rather, excuse the pun, strange issue. In my game I have two different contexts 

1. File management: loads and saves different states from file, this includes the player's state and inventory
2. The game context: everything else regarding the game goes here. 

In the file management context, I am injecting a player model(actually just a wrapper that contains different models related to the player's health, powers, speed, etc) and making it a singleton that spans accross contexts(crossContext()). Once file loading routines are completed, the game goes right into loading a specified level,models of course are again loaded from file and is binded to a singleton. A signal is dispatched in this context to Instantiate a player prefab[figure1] and then sets its parent to one of the children in the game context hierarchy. This is where I start to run into an issue: One of the views attached to the player is a PlayerControllerView and its respective mediator " PlayerControllerViewMediator" is added twice in the game object. I have searched my code to ensure that there isn't any double binding shenanigans and that doesn't seem to be the case. What could be the probably cause behind this? 
 
![Figure 1](https://cloud.githubusercontent.com/assets/1048661/9510461/e51ea15c-4c39-11e5-8f68-d6fd4bdb51b5.png)

![Figure 2](https://cloud.githubusercontent.com/assets/1048661/9510465/f685080a-4c39-11e5-897e-29282a9b6cda.png)

**Comment by mohaider** [comment]
I just wanted to provide more information on the matter as I do believe it is a bug as I was able to reproduce it in another project with some very simple views, empty classes whose only function is to inherit from view and their respective mediators throw in the mix.   I should also mention that this appears to be an issue with instantiating a prefab  using   GameObject.Instantiate(UnityEngine.Resources.Load(TestPrefabPath)) as
                    GameObject; 

I am not sure if this could be of any help but the version of unity I am running is 5.1.2f1
Furthermore after debugging, I believe I found the possible cause of this problem. 
Creating a new gameobject and adding their individual components doesn't appear to cause an issue which brings me to this hypothesis. I believe the cause of this problem could be in the way that the mediationBinder is handling gameobjects that contain multiple views.  

  
I looked at the call stack during the bubble up phase, the initial view appears to be binding its mediator twice. The first time by subroutines invoked during  injectViewAndChildren, once completed by mapView. Trigger cs:53
